### PR TITLE
Fix deployment, change iterator and implement issue #4.

### LINKED ISSUE #3: Memory leak when iterating over a db.
When iterating over a large database and accesding each field of each event, the programm quickly consumes gigabytes of RAM. The problem may be located in the JNI cursor_next method.

### LINKED ISSUE #4: Access field values in TrailDBEvent.
Create again a method to access the field values of an event.

### PR DESCRIPTION
**Pull request by Nennya**
Provide way to access field values of an event. Closes #4 .
The TrailDBIterator has to be used in a try-with-resource block to prevent memory leaks. Closes #3 .
Quick update of the travils config to avoid deployment on PR.

### CODE REVIEW

**Comment by ChrisJamesC** [comment]
Please give a more explicit name to the PR and add "Closes #4" to the description, so it can close the issue automatically when the PR is merged.

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
       install: true
```
Comment (ChrisJamesC) [inline .travis.yml]: This change belongs to a separate PR.

File: src/main/java/io/sqooba/traildb/TrailDBIterator.java [hunk 2]
```diff
@@ -52,6 +52,9 @@ public TrailDBEvent next() {
             public boolean hasNext() {
                 TrailDBEvent next = TrailDBNative.INSTANCE.cursorNext(TrailDBIterator.this.cursor);
                 if (next == null) {
+                    // Auto-close when at the end because it is not intuitive to have to close an
```
Comment (ChrisJamesC) [inline src/main/java/io/sqooba/traildb/TrailDBIterator.java]: Isn't it issue #3 instead of #4?

Comment (debelyoo) [inline src/main/java/io/sqooba/traildb/TrailDBIterator.java]: Does this fix the leak you had (i.e. issue #3 ) ?

File: src/main/java/io/sqooba/traildb/TrailDBIterator.java [hunk 3]
```diff
@@ -52,6 +52,9 @@ public TrailDBEvent next() {
...
+                    // iterator when at its end in Java.
+                    TrailDBIterator.this.close();
```
Comment (debelyoo) [inline src/main/java/io/sqooba/traildb/TrailDBIterator.java]: Does this fix the leak you had (i.e. issue #3 ) ?
BTW, I'm not really understanding your comment... what do you mean by "at the end" ? at the end of the iteration process ?

Comment (Nennya) [inline src/main/java/io/sqooba/traildb/TrailDBIterator.java]: The issue was due to the fact I was not manually closing each iterator, I had the same problem when reading but forgot.
Yes when we have iterated over all events in the trail.

Comment (debelyoo) [inline src/main/java/io/sqooba/traildb/TrailDBIterator.java]: Ok, I would still rephrase it a bit... something like "Auto-close the iterator when it reaches its end. Do this explicitly because, in Java, it is not intuitive to close an iterator that reaches its end".
And indeed this auto-close is not intuitive to me :) Shouldn't you do it on the next() method as well then ? 
But I also see that the TrailDBIterator implements AutoCloseable. Couldn't we force (or at least encourage) the users to use it with a try-with-resources statement ? The users should know how to use the structure they use. If it implements autocloseable, I would not force a close when the iterator reaches its end.

Comment (Nennya) [inline src/main/java/io/sqooba/traildb/TrailDBIterator.java]: I will rephrase.
We have to close it only if we are done because then we can not use it anymore, so if we do it in the next method, a subsequent call to next would fail.
I agree that the current state of the iterator is not great. It has been implemented in a manner that it reflects what is done in the C code but at the same time I tried to make it like a Java iterator and the two seem to not work well together.
